src/main/java/software/amazon/qldb/QldbDriverImplBuilder.java

```diff
@@ -158,12 +159,14 @@ private QldbDriver createDriver() {
                 oc.retryPolicy(software.amazon.awssdk.core.retry.RetryPolicy.builder().numRetries(0).build());
             });
         });
-        AttributeMap httpConfig = AttributeMap
-            .builder()
-            .put(SdkHttpConfigurationOption.MAX_CONNECTIONS, maxConcurrentTransactions)
-            .build();
+        if (maxConcurrentTransactions != DEFAULT_MAX_CONCURRENT_TRANSACTIONS) {
```

Comment:
It seems odd that you have an inconsistent interaction here. If you set maxConcurrrentTxns then you lose your setting. 

If possible, look into if it's able to modify or combine the http client settings that already exist, and overwrite just the MAX_CONNECTIONS.